packages/aragon-wrapper/src/index.js

```
@@ -445,6 +449,81 @@ export default class Aragon {
+      }
+      return Promise.resolve()
+    }))
+
+    this.identityProviderRegistrar = new Map(
+      identityProviders.map(({ name, provider }) => [name, provider])
+    )
+
+    // Set up identity modification intent observable
+    this.addressIdentityModificationIntents = new Subject()
+  }
+
+  /**
+   * Modify the identity metadata for an address using the requested provider.
+   *
+   * @param  {string} address Address to modify
+   * @param  {Object} metadata Modification metadata object
+   * @param  {string} [providerName='local'] Name of the identity provider to use
+   * @return {void}
+   */
+  async modifyAddressIdentity (address, metadata, providerName = 'local') {
+    const provider = this.identityProviderRegistrar.get(providerName)
+    if (provider && typeof provider.resolve === 'function') {
+      return provider.modify(address, metadata)
+    }
+    throw new Error(`Provider (${providerName}) not installed`)
+  }
+
+  /**
+   * Resolve the identity metadata for an address using the requested provider.
+   *
+   * @param  {string} address Address to resolve
+   * @param  {string} [providerName='local'] Name of the identity provider to use
+   * @return {void}
+   */
+  async resolveAddressIdentity (address, providerName = 'local') {
+    const provider = this.identityProviderRegistrar.get(providerName)
+    if (provider && typeof provider.resolve === 'function') {
+      return provider.resolve(address)
+    }
+    throw new Error(`Provider (${providerName}) not installed`)
+  }
+
+  /**
+   * Request an identity modification using the requested provider.
+   *
+   * @param  {string} address Address to modify
+   * @param  {string} [providerName='local'] Name of the identity provider to use
+   * @return {void}
+   */
+  async requestAddressIdentityModification (address, providerName) {
+    if (this.identityProviderRegistrar.has(providerName)) {
+      this.addressIdentityModificationIntents.next({ address, providerName })
+    }
```

Not sure if we should error out if it's not available, but probably.